core/src/test/java/org/infinispan/functional/AbstractFunctionalOpTest.java

```diff
@@ -283,9 +283,9 @@ public static VisitableCommand getCurrent() {
       }
 
       @Override
-      protected BasicInvocationStage handleDefault(InvocationContext ctx, VisitableCommand command) throws Throwable {
+      protected InvocationStage handleDefault(InvocationContext ctx, VisitableCommand command) throws Throwable {
          current.set(command);
-         return invokeNext(ctx, command).handle((rCtx, rCommand, rv, t) -> current.remove());
+         return invokeNext(ctx, command).whenComplete(ctx, command, (rCtx, rCommand, rv, t) -> current.remove());
```

Comment:
That's just a `finally` handler, I think that previously the `handle()` method was called incorrectly (since it accepted a consumer, while `CF.handle()` accepts a function), so it should be correct here.